src/WebJobs.Extensions.DurableTask/ActorScheduler/SchedulerState.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using DurableTask.Core.Serializing;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// The persisted state of an actor scheduler, as handed forward between ContinueAsNew instances.
+    /// </summary>
+    [JsonConverter(typeof(CustomJsonConverter<SchedulerState>))]
+    internal class SchedulerState : CustomJsonConverter<SchedulerState>.IConversion
+    {
+        /// <summary>
+        /// Whether this actor exists or not.
+        /// </summary>
+        public bool ActorExists { get; set; }
+
+        /// <summary>
+        /// The serialized actor state. This can be stale while CurrentStateView != null.
+        /// </summary>
+        public string ActorState { get; set; }
+
+        /// <summary>
+        /// The queue of waiting operations, or null if none.
+        /// </summary>
+        public Queue<OperationMessage> Queue { get; set; }
+
+        [JsonIgnore]
+        public bool IsEmpty => !ActorExists && (Queue == null || Queue.Count == 0);
+
+        [JsonIgnore]
+        internal IStateView CurrentStateView { get; set; }
+
+        internal void Enqueue(OperationMessage operationMessage)
+        {
+            if (this.Queue == null)
+            {
+                this.Queue = new Queue<OperationMessage>();
+            }
+
+            this.Queue.Enqueue(operationMessage);
+        }
+
+        internal bool TryDequeue(out OperationMessage operationMessage)
+        {
+            operationMessage = null;
+
+            if (this.Queue == null)
+            {
+                return false;
+            }
+
+            operationMessage = this.Queue.Dequeue();
+
+            if (this.Queue.Count == 0)
+            {
+                this.Queue = null;
+            }
+
+            return true;
+        }
+
+        public override string ToString()
+        {
+            return $"state.length={((this.ActorState != null) ? this.ActorState.Length : 0)} queue.count={((this.Queue != null) ? this.Queue.Count : 0)}";
+        }
+
+        public void FromJson(JsonReader reader, JsonSerializer serializer)
```

Comment:
A couple suggestions regarding custom serialization logic:
* I suggest we add a comment explaining why we use custom serialization instead of the reflection-based serialization that most people will expect.
* We should also add a comment giving an example of what the JSON payload we're parsing is expected to look like. That will make human validation of the code easier.